haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
+
+
+class SentenceTransformersEmbeddingBackend:
+    """
+    Singleton class to manage SentenceTransformers embeddings.
+    """
+
+    _instances: Dict[str, "SentenceTransformersEmbeddingBackend"] = {}
+
+    def __new__(cls, *args, **kwargs):
+        args_kwargs_str = str(args) + str(kwargs)
+        instance_id = hashlib.md5(args_kwargs_str.encode())
+
+        if instance_id in cls._instances:
+            return cls._instances[instance_id]
+
+        instance = super().__new__(cls)
+        cls._instances[instance_id] = instance
+        return instance
+
+    def __init__(
+        self, model_name_or_path: str, device: Optional[str] = None, use_auth_token: Union[bool, str, None] = None
+    ):
+        sentence_transformers_import.check()
+        if not hasattr(self, "model"):
+            self.model = SentenceTransformer(
+                model_name_or_path=model_name_or_path, device=device, use_auth_token=use_auth_token
+            )
+
+    def embed(self, data: List[str], *inference_params) -> np.ndarray:
```

Comment:
I think it makes sense only if we're using them in the method, for example to validate them. Otherwise if we're just passing them over untouched I think it's very convenient to use kwargs.

Reply:
Can we make it `**kwargs`?